soh/soh/Enhancements/TimeSavers/SkipCutscene/SkipIntro.cpp

```diff
@@ -45,3 +45,5 @@ void SkipIntro_Register() {
         }
     });
 }
+
+static RegisterShipInitFunc initFunc(RegisterSkipIntro);
```

Comment:
This should probably be

```suggestion
static RegisterShipInitFunc initFunc(RegisterSkipIntro, { CVAR_ENHANCEMENT("TimeSavers.SkipCutscene.Intro"), "IS_RANDO" });
```

So that this initFunc is only called when either the relevant CVar is changed or when a rando save is loaded (we need both for the default IS_RANDO value to work properly). Otherwise it will run this initFunc once on boot and once every time any CVar is changed. You _might_ also be able to do a `COND_VB_SHOULD` above this but I'm not 100% sure if that would work.

Reply:
The logic was complicated enough that I left cvar / setting checks in the main code. My understanding is that these rerun items are for dependencies to `COND` macro conditions, which `RegisterSkipIntro` is not using. ie there is no need to register/unregister it when it's registered unconditionally

Reply:
I don't think they're supposed to be registered unconditionally.

Reply:
Pushed version with `shouldRegister` to demonstrate my point. You still need to check cvar inside hook because there's a weird case where rando forces it sometimes (adult start & entrance shuffle)

This could be much simpler if rando just always skipped intro

Reply:
To be honest, it's uglier, but I still prefer the explicit declarations of what exactly is going on, so we'll stick with that.